src/fflogs.ts

```diff
@@ -0,0 +1,228 @@
+import {AxiosRequestConfig} from 'axios'
+
+// -----
+// Fight
+// -----
+
+export interface Fight {
+	id: number
+
+	boss: number
+	name: string
+	zoneID: number
+	zoneName: string
+	lastPhaseForPercentageDisplay?: number
+
+	difficulty?: number
+	bossPercentage?: number
+	fightPercentage?: number
+	kill?: boolean
+	partial?: number
+	size?: number
+	standardComposition?: boolean
+
+	start_time: number
+	end_time: number
+}
+
+export interface Phase {
+	boss: Fight['boss']
+	phases: string[]
+}
+
+// -----
+// Actors
+// -----
+
+export enum ActorType {
+	// Enemy
+	BOSS = 'Boss',
+	NPC = 'NPC',
+
+	// Friendly
+	PALADIN = 'Paladin',
+	WARRIOR = 'Warrior',
+	DARK_KNIGHT = 'DarkKnight',
+	WHITE_MAGE = 'WhiteMage',
+	SCHOLAR = 'Scholar',
+	ASTROLOGIAN = 'Astrologian',
+	MONK = 'Monk',
+	DRAGOON = 'Dragoon',
+	NINJA = 'Ninja',
+	SAMURAI = 'Samurai',
+	BARD = 'Bard',
+	MACHINIST = 'Machinist',
+	BLACK_MAGE = 'BlackMage',
+	SUMMONER = 'Summoner',
+	RED_MAGE = 'RedMage',
+	LIMIT_BREAK = 'LimitBreak',
+
+	// Pet
+	PET = 'Pet'
+}
+
+export interface ActorFightInstance {
+	id: Fight['id'],
+	groups?: number,
+	instances?: number,
+}
+
+export interface Actor {
+	guid: number
+	id: number
+	name: string
+	type: ActorType
+	fights: ActorFightInstance[]
+}
+
+export interface Pet extends Actor {
+	petOwner: Actor['id']
+}
+
+export interface ActorResources {
+	hitPoints: number
+	maxHitPoints: number
+	mp: number
+	maxMP: number
+	tp: number
+	maxTp: number
+	x: number
+	y: number
+}
+
+// -----
+// Events
+// -----
+
+export enum AbilityType {
+	PHYSICAL_DOT = 1,
+	HEAL = 8,
+	SPECIAL = 32,
+	MAGICAL_DOT = 64,
+	PHYSICAL_DIRECT = 128,
+	LIMIT_BREAK = 256,
+	MAGICAL_DIRECT = 1024,
+}
```

Comment:
I discussing with the snek if these shouldn't be hex flags instead (0x1, 0x4, 0x20, 0x40, 0x80, 0x100, 0x400) but it seems fflogs' UI doesn't support hex 🤦‍♀️